drivers/source/serial_io.c

```diff
@@ -19,42 +19,50 @@
 
 /* USER CODE BEGIN */
 // Add a static assertion to ensure that UART_PRINT_REG is defined as either scilinREG or sciREG
+STATIC_ASSERT(UART_PRINT_REG == sciREG || UART_PRINT_REG == scilinREG, "UART_PRINT_REG should be set to sciREG or scilinREG");
 
 /* USER CODE END */
 STATIC_ASSERT(MAX_PRINTF_SIZE > 0, "MAX_PRINTF_SIZE must be greater than 0");
 
 /* GLOBAL VARIABLES */
-static SemaphoreHandle_t sciMutex; // Protects SCI1/SCI module
-static StaticSemaphore_t sciMutexBuffer; // Buffer for SCI mutex
+static SemaphoreHandle_t sciMutex = NULL; // Protects SCI1/SCI module
+static StaticSemaphore_t sciMutexBuffer;  // Buffer for SCI mutex
 
-static SemaphoreHandle_t sciLinMutex; // Protects SCI2/SCILin module
-static StaticSemaphore_t sciLinMutexBuffer; // Buffer for SCI2 mutex
+static SemaphoreHandle_t sciLinMutex = NULL; // Protects SCI2/SCILin module
+static StaticSemaphore_t sciLinMutexBuffer;  // Buffer for SCI2 mutex
 
 /**
  * @brief Iterate through an array of bytes and transmit them via UART_PRINT_REG.
- * 
+ *
  * @param bytes The array of bytes to transmit.
  * @param length The length of the array of bytes to transmit.
  * @return OBC_ERR_CODE_SUCCESS on success, else an error code
  */
 static obc_error_code_t sciSendBytes(unsigned char *bytes, uint32_t length);
 
-void sciMutexInit(void) {
-    if (sciMutex == NULL) {
+void sciMutexInit(void)
+{
+    if (sciMutex == NULL)
+    {
         sciMutex = xSemaphoreCreateMutexStatic(&sciMutexBuffer);
     }
-    
+
     /* USER CODE BEGIN */
     // Create mutex to protect SCI2/SCILin module here.
-    
+    if (sciLinMutex == NULL)
+    {
+        sciLinMutex = xSemaphoreCreateMutexStatic(&sciLinMutexBuffer);
+    }
+
     /* USER CODE END */
 
     // Static allocation of mutexes can only fail if the buffer is NULL
     configASSERT(sciMutex != NULL);
     configASSERT(sciLinMutex != NULL);
 }
 
-obc_error_code_t sciPrintText(unsigned char *text, uint32_t length) {
+obc_error_code_t sciPrintText(unsigned char *text, uint32_t length)
+{
```

Comment:
We usually put the opening curly brace (`{`) on the same line as the function declaration. No need to change this for the onboarding tho.